connectors/module-messageStore/messageStore/message_store_endpoint.bal

```diff
@@ -0,0 +1,244 @@
+                            + self.queueName, err = storeResult);
+                            reTrySuccessful = false;
+                        } else {
+                            //send successful
+                            break;
+                        }
+                    }
+                    if (!reTrySuccessful) {
+                        int retryDelay = retryConfig.interval
+                        + math:round(retryCount * retryConfig.interval * retryConfig.backOffFactor);
+                        if (retryDelay > retryConfig.maxWaitInterval) {
+                            retryDelay = retryConfig.maxWaitInterval;
+                        }
+                        runtime:sleep(retryDelay);
+                        retryCount = retryCount + 1;
+                    }
+                }
+
+                //if max retries breached. Check for failover store
+                if (retryCount == retryConfig.count) {
+                    Client? failoverClient = self.failoverStore;
+                    //try failover store
+                    if (failoverClient is Client) {
+                        var failOverClientStoreResult = failoverClient->store(request);
+                        if (failOverClientStoreResult is error) {
+                            log:printError("Error while sending message to failover store. Message store queue = "
+                            + self.queueName, err = failOverClientStoreResult);
+                            return failOverClientStoreResult;
+                        }
+                    } else {
+                        //if no failover store, return original store error
+                        return storeSendResult;
+                    }
+                }
+            }
+        }
+    }
+
+
+    # Try to deliver the message to message broker queue.  
+    #
+    # + requestMessageMap - Map representation `map<string>` of the message to store  
+    # + return - `error` in case of an issue delivering the message to the queue
+    function tryToSendMessage(map<string> requestMessageMap) returns error? {
+        //create a map message from message detail extracted
+        //TODO: here if error occurs it is not returned as an error. Ballerina should be fixed. (/ballerina-lang/issues/16099)
+        var messageToStore = self.jmsSession.createMapMessage(requestMessageMap);
+        io:println("is error: " + (messageToStore is error));
+        if (messageToStore is jms:Message) {
+            io:println("##############" + true);
+            // This sends the Ballerina message to the JMS provider.
+            var returnVal = self.queueSender->send(messageToStore);
+            if (returnVal is error) {
+                string errorMessage = "Error occurred while sending the message to the queue " + self.queueName;
+                log:printError(errorMessage, err = returnVal);
+            }
+        } else {
+            io:println("##############" + false);
```

Comment:
Is this a test println? Shall we remove it or add a proper log message?

Reply:
I will remove it. Added for a test.